src/rpc/net.cpp

```diff
@@ -322,6 +323,61 @@ static RPCHelpMan addnode()
     };
 }
 
+static RPCHelpMan addconnection()
```

Comment:
hmmm, I see that `addnode` RPC returns this error, but seems like that's what is indicated by the bool that the `AddNode` function returns since it's comparing to `vAddedNodes`. In the current implementation, we don't currently know that information until we call through to `OpenNetworkConnection`, which simply returns if we are  `AlreadyConnectedToAddress(addrConnect))`. We could add a call from `AddConnection` to check that and return early. And then it would have to populate a boolean as an in/out param (since the existing return value indicates whether there is sufficient capacity). This is all doable, but I'm not really seeing the additional value. It would be able to catch a bug in the test framework logic I'm adding, but that bug would also be caught by the tests in `p2p_add_connections.py`. And I don't think this would be helpful for additional tests that use the test framework utilities? What do you think?

Reply:
Similar to https://github.com/bitcoin/bitcoin/pull/19315#discussion_r525435591, I think more detailed error reporting can be done in a follow-up PR.